src/utils/form/replaceField.ts

```diff
@@ -1,29 +1,44 @@
 import { getQuestion } from './getQuestion';
+import isEqual from 'lodash/isEqual';
 
 /**
- * Check if the structure is correct and replace the passed field with the corresponding one from the passed structure.
+ * Check if the structure is correct and replace the chosen field by the corresponding one in the referenceStructure.
  * Function by induction.
- * @param structure structure of the form to edit
- * @param name name of the field to search for
- * @param template 
+ * @param fieldName name of the field to search for
+ * @param editedStructure structure of the form that will be edited
+ * @param referenceStructure structure which should be used as a reference to change field value
+ * @param prevReferenceStructure structure which represent the previous state of the reference structure
  * @returns 
  */
-export const replaceField = (structure: any, name: string, template: any): boolean => {
+export const replaceField = (fieldName: string, editedStructure: any, referenceStructure: any, prevReferenceStructure: any): boolean => {
   // Loop on elements to find the right question
-  if (structure.pages) {
-    for (const page of structure.pages) {
-      if (replaceField(page, name, template)) return true;
+  if (editedStructure.pages) {
+    for (const page of editedStructure.pages) {
+      if (replaceField(fieldName, page, referenceStructure, prevReferenceStructure)) return true;
     }
-  } else if (structure.elements) {
-    for (const elementIndex in structure.elements) {
-      const element = structure.elements[elementIndex];
-      if (element.type === 'panel') {
-        if (replaceField(element, name, template)) return true;
-      } else {
-        if (element.valueName === name) {
-          // Replace the field
-          structure.elements[elementIndex] = getQuestion(template, name);
-          return true;
+  } else if (editedStructure.elements) {
+    for (const elementIndex in editedStructure.elements) {
+      const element = editedStructure.elements[elementIndex];
+      if (element) {
+        if (element.type === 'panel') {
+          if (replaceField(fieldName, element, referenceStructure, prevReferenceStructure)) return true;
+        } else {
+          if (element.valueName === fieldName) {
+            const referenceField = getQuestion(referenceStructure, fieldName);
+            const prevReferenceField = getQuestion(prevReferenceStructure, fieldName);
+            // If the edited structure's field has a defaultValue, and this defaultValue
+            // isn't equal to the previous version of the reference structure's field's defaultValue
+            if (element.hasOwnProperty('defaultValue') && !isEqual(element.defaultValue, prevReferenceField?.defaultValue)) {
```

Comment:
@Matthis-M-ReliefApps a good to know with Javascript, previously it was el.default !== prev.default, but it cannot work with non scalar values. isEqual is quite powerful and better handles equality between objects